realm/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -2058,4 +2058,78 @@ private void awaitOrFail(CountDownLatch latch) {
             fail();
         }
     }
+
+    private void populateForDistinct(Realm realm, long numberOfBlocks, long numberOfObjects) {
+        realm.beginTransaction();
+        for (int i = 0; i < numberOfObjects * numberOfBlocks; i++) {
+            for (int j = 0; j < numberOfBlocks; j++) {
+                AnnotationIndexTypes obj = realm.createObject(AnnotationIndexTypes.class);
+                obj.setIndexBoolean(j % 2 == 0);
+                obj.setIndexLong(j);
+                obj.setIndexDate(new Date(1000 * j));
+                obj.setIndexString("Test " + j);
+                obj.setNotIndexBoolean(j % 2 == 0);
+                obj.setNotIndexLong(j);
+                obj.setNotIndexDate(new Date(1000 * j));
+                obj.setNotIndexString("Test " + j);
+            }
+        }
+        realm.commitTransaction();
+    }
```

Comment:
One more test case for `null`? Does the `distinct` work on same rows with `null` values as well?